src/braket_container.py

```diff
@@ -301,12 +305,10 @@ def run_customer_code() -> None:
     local_s3_file = download_customer_code(s3_uri)
     unpack_code_and_add_to_path(local_s3_file, compression_type)
     install_additional_requirements()
-    if entry_point.find(":") >= 0:
-        exit_code = run_customer_code_as_process(entry_point)
-    else:
-        exit_code = run_customer_code_as_subprocess(entry_point)
-    if exit_code != 0:
-        log_failure_and_exit(f"Job at {entry_point} exited with exit code: {exit_code}")
+    customer_method = extract_customer_method(entry_point)
+    customer_method_process = kick_off_customer_script(customer_method)
+    join_customer_script(customer_method_process)
+    sys.exit(customer_method_process.exitcode)
```

Comment:
I'm not crazy about exiting with 0 here, since we may want to add additional functionality after this method. I would check if it's non-zero (as we were doing before), and then exit with 0 (or SM won't propagate the error correctly). Or we should check if they've fixed that bug.

Reply:
Yup, still see a blank sm error message with a non-zero exit code. Fixed to return zero on failure as before

Reply:
Tbf, there's no sagemaker error when exiting with zero anyways. The braket failure reason does get propagated in both cases, just a difference in the logs. I'd argue perhaps `SM_ERROR=""` is better than `Reporting training SUCCESS`, so maybe we want to use the correct exit code?